README.md

```diff
@@ -91,122 +91,195 @@ It is recommended to build the code in a separate directory from the source dire
+
+Running the help of the `domain` tool gives the following:
+
 ```
-mpirun -n 2 ./decomp --grid grid.nc --dim0 y --dim1 x --mask land_mask
+$ ./build/decomp -h
+Usage: ./build/decomp [options]
+Options:
+  -h [ --help ]             Display this help message
+  -g [ --grid ] arg         NetCDF grid file
+  -x [ --xdim ] arg (=x)    Name of x dimension in netCDF grid file
+  -y [ --ydim ] arg (=y)    Name of y dimension in netCDF grid file
+  -o [ --order ] arg (=yx)  Order of dimensions in netCDF grid file, e.g., 'yx'
+                            or 'xy'
+  -m [ --mask ] arg (=mask) Mask variable name in netCDF grid file
+  -i [ --ignore-mask ]      Ignore mask in netCDF grid file
+
 ```
 
-The `decomp` tool produces two netCDF-4 files (using the classic data model) named `partition_mask_<num_mpi_processes>.nc` and `partition_metadata_<num_mpi_processes>.nc` with the following layout:
+We can see that the most of the options have defaults (show in parentheses) e.g., if you do not specify the name of the land
+mask `-m/--mask`, it will default to `"mask"`. To explain the other options it is instructive to first look at an example grid
+file.
 
 ```
-netcdf partition_mask_2 {
+$ ncdump test_2.nc
+netcdf test_2 {
 dimensions:
-	x = 30 ;
-	y = 30 ;
+        m = 6 ;
+        n = 4 ;
 variables:
-	short pid(x, y) ;
+        int land_mask(n, m) ;
 
 // global attributes:
-		:num_processes = 2s ;
+                :title = "Non-default dimension naming and ordering" ;
 data:
 
- pid = ...
+ land_mask =
+  0, 0, 0, 0, 0, 0,
+  1, 1, 1, 1, 1, 1,
+  0, 0, 0, 0, 0, 0,
+  1, 1, 1, 1, 1, 1 ;
 }
 ```
 
-The netCDF variable `pid` is defined as the process ID of each point in the grid.
+This is from a grid file `test_2.nc` where the `x` and `y` dimensions are called `m` and `n` respectively. Looking at the
+`variables` sections we can see that the land mask `land_mask` is stored in the order `land_mask(n, m)`, where `n` is the first
+dimension. No matter what the dimensions are called inside of the grid file, neXtSIM_DG will expect the metadata files (produced
+by `decomp`) to have a specific output format. By default, the ordering of `decomp` assumes `'yx'` which (due to the names of our
+dimensions) corresponds to `n` being the first dimension and `m` being the second. This can be changed to `'xy'` for example, if
+the land mask was stored transposed i.e., `land_mask(m, n)`.
 
+To run `decomp` on the `test_2.nc` file we can use the following command:
+```
+$ mpirun -n 2 ../decomp -g test_2.nc -x m -y n -o 'yx' -m "land_mask"
```

Comment:
More generally it would be `/path/to/decomp`, also noting that `../decomp` already assumes you are in the build dir.

Reply:
thanks done in [cea56b4](https://github.com/nextsimhub/domain_decomp/pull/58/commits/cea56b4d24313d498c7648ddbd8b6e489cee0311)